sky/serve/controller.py

```diff
@@ -126,6 +126,21 @@ async def update_service(request: fastapi.Request):
                              f'{common_utils.format_exception(e)}')
                 return {'message': 'Error'}
 
+        @self._app.post('/controller/terminate_replica')
+        async def terminate_replica(request: fastapi.Request):
+            request_data = await request.json()
+            try:
+                replica_id = request_data.get('replica_id')
+                if replica_id is None:
+                    return {'message': 'Error: Replica ID is not specified.'}
+                logger.info(f'Terminating replica {replica_id}')
+                self._replica_manager.scale_down(replica_id)
```

Comment:
It is not enough to only call `replica_manager.scale_down` since one important usecase of this command is cleanup failed records, but this function will keep any failed record. Should we add a `purge` argument to force clean it from database?

Reply:
Is something like this change (https://github.com/skypilot-org/skypilot/pull/3179/commits/b838b1f23565574084ee80da402933fb83fcab1c) what you had in mind? If so, what's the best way to test this?

Reply:
Yes it is, except for we need to terminate the cluster too when purging. IIUC current implementation will only remove the replica record. You can try to `sky serve down --purge <service-name> -r <id>` for : 1. a failed replica, 2. a ready replica and see if the record and cluster got removed.

Reply:
Also, we should print some error message if we terminate a failed replica without `--purge` and prompt the user to use `--purge` to cleanup the record.